go/private/rules/cgo.bzl

```diff
@@ -382,13 +429,54 @@ def setup_cgo_library(name, srcs, cdeps, copts, clinkopts):
       visibility = ["//visibility:private"],
   )
 
+  cgo_cxx_lib_name = name + ".cgo_cxx_lib"
+  native.cc_library(
+      name = cgo_cxx_lib_name,
+      srcs = [select_cxx_files],
+      deps = cdeps,
+      copts = cxxopts + platform_copts + [
+          # The generated thunks often contain unused variables.
+          "-Wno-unused-variable",
+      ],
+      linkopts = clinkopts + platform_linkopts,
+      linkstatic = 1,
+      # _cgo_.o needs all symbols because _cgo_import needs to see them.
+      alwayslink = 1,
+      visibility = ["//visibility:private"],
+  )
+
+  cgo_o_deps = [
+      cgo_c_lib_name,
+      cgo_cxx_lib_name,
+  ]
+  cgo_collect_info_libs = cgo_o_deps[:]
+
+  if objc:
+    cgo_objc_lib_name = name + ".cgo_objc_lib"
+    native.objc_library(
+        name = cgo_objc_lib_name,
+        srcs = [select_objc_files],
+        deps = cdeps,
+        copts = copts + platform_copts + [
+            # The generated thunks often contain unused variables.
+            "-Wno-unused-variable",
+        ],
+        # _cgo_.o needs all symbols because _cgo_import needs to see them.
+        alwayslink = 1,
+        visibility = ["//visibility:private"],
+        **objcopts
+    )
+    cgo_o_deps.append(cgo_objc_lib_name)
+    # Use the implicit fully linked target
+    cgo_collect_info_libs.append(cgo_objc_lib_name + "_fully_linked.a")
```

Comment:
This seems like it could easily break in the future.

For the `cc_binary` rule below, is it possible to use `apple_binary` instead (when `objc = True`)? This might have other negative implications though; I'm not sure.

Reply:
Good call. I kind of remember having tried that before, but I don't quite recall why I've moved away from it. I'll try again and let you know.

Reply:
So I retested it: it's because of the `platform_type` argument to `apple_binary`. It's not possible in a macro to know what platform is built...

Reply:
Ok, too bad that doesn't work.

Could you expand on the comment here why `_fully_linked.a` is used? I'm assuming that includes transitive dependencies, but it would be good to document that.